command/catalog_nodes.go

```diff
@@ -0,0 +1,143 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/configutil"
+)
+
+// CatalogNodesCommand is a Command implementation that is used to fetch all the
+// datacenters the agent knows about.
+type CatalogNodesCommand struct {
+	BaseCommand
+}
+
+func (c *CatalogNodesCommand) Help() string {
+	helpText := `
+Usage: consul catalog datacenters [options]
+
+  Retrieves the list nodes registered in a given datacenter. By default, the
+  datacenter of the local agent is queried.
+
+  To retrieve the list of nodes:
+
+      $ consul catalog nodes
+
+  For a full list of options and examples, please see the Consul documentation.
+
+` + c.BaseCommand.Help()
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *CatalogNodesCommand) Run(args []string) int {
```

Comment:
I was kind of expecting `consul members`-like table output for this, too, since there's a lot of info vs `consul kv get -detailed`. Maybe something like:

1. `consul catalog node list` maybe a table of: node name, node id, address
2. `consul catalog node info -node=<name or id>` does the `kv -detailed` kind of thing

Still kind of thinking about this one...